frame/staking/src/lib.rs

```diff
@@ -1654,6 +1665,12 @@ decl_module! {
 		///
 		/// The dispatch origin must be Root.
 		///
+		/// # Warning
+		///
+		/// The election process start multiple block before the end of the era.
+		/// Thus the election process may have started when this is called. In this case the
+		/// election will be ongoing until the next era is triggered.
+		///
```

Comment:
If we change to `ForceNone`, does that mean a new era will not be triggered until we change forcing again? Or does `ForceNone` only come into affect for the next era?

> election will be ongoing until the next era is triggered.

If it is the middle of an election and we call `force_no_eras`, would the election keep going indefinitely? (This is what the comment suggests to me when I read it) Or should an election always end at some point even if era does not change?

Reply:
> If we change to `ForceNone`, does that mean a new era will not be triggered until we change forcing again? Or does `ForceNone` effects only come into affect for the next era?

no it is permanent, until the forcing is changed again.

> > election will be ongoing until the next era is triggered.
> 
> If it is the middle of an election and we call `force_no_eras`, would the election keep going indefinitely? (This is what the comment suggests to me when I read it) Or should an election always end at some point even if era does not change?

yes, this is what is happening, the election is ongoing indefinitely.